### PR TITLE
Make cart aware of customer

### PR DESCRIPTION
**Pull request by wychoong**
- [x] Documentation updates
- [x] Automated tests
- [x] Changelog entries (core and hub)

### CODE REVIEW

File: packages/core/src/Managers/StorefrontSessionManager.php [hunk 1]
```diff
@@ -32,6 +34,13 @@ class StorefrontSessionManager implements StorefrontSessionInterface
      */
     protected ?Currency $currency = null;
 
+    /**
+     * The current currency
```
Comment (wychoong) [inline packages/core/src/Managers/StorefrontSessionManager.php:38]:
```suggestion
     * The current customer
```

File: packages/core/src/Managers/StorefrontSessionManager.php [hunk 2]
```diff
@@ -122,6 +132,38 @@ public function initChannel()
...
+        $customer_id = $this->sessionManager->get(
+            $this->getSessionKey().'_customer'
+        );
+
+        if (! $customer_id) {
+            if (Auth::check() && is_lunar_user(Auth::user())) {
+                $user = Auth::user();
+
+                if ($customer = $user->customers()->orderBy('created_at', 'desc')->orderBy('id', 'desc')->first()) {
+                    $customer_id = $customer->id;
```
Comment (wychoong) [inline packages/core/src/Managers/StorefrontSessionManager.php:153]:
```suggestion
                    return $this->setCustomer($customer);
```

File: packages/core/src/Base/StorefrontSessionInterface.php [hunk 3]
```diff
@@ -71,5 +72,18 @@ public function getChannel(): Channel;
...
+     * @return self
+     */
+    public function setCustomer(Customer $customer): self;
+    
+    /**
+     * Return the current customer.
+     *
+     * @return \Lunar\Models\Customer
+     */
+    public function getCustomer(): Customer;
```
Comment (wychoong) [inline packages/core/src/Base/StorefrontSessionInterface.php:88]:
```suggestion
     * @return \Lunar\Models\Customer | null
     */
    public function getCustomer(): ?Customer;
```

**Comment by alecritson** [comment]
@wychoong Converting to draft until it's complete and ready for full review.

**Comment by glennjacobs** [comment]
@wychoong could you target `main` for this PR please?

**Comment by wychoong** [comment]
closing this with #1125

**Pull request closed without merging**